internal/impl/splunk/output.go

```diff
@@ -0,0 +1,203 @@
+		return
+	}
+
+	if o.useGzipCompression, err = pConf.FieldBool(soFieldGzip); err != nil {
+		return
+	}
+
+	if o.eventHost, err = pConf.FieldString(soFieldEventHost); err != nil {
+		return
+	}
+
+	if o.eventSource, err = pConf.FieldString(soFieldEventSource); err != nil {
+		return
+	}
+
+	if o.eventSourceType, err = pConf.FieldString(soFieldEventSourceType); err != nil {
+		return
+	}
+
+	if o.eventIndex, err = pConf.FieldString(soFieldEventIndex); err != nil {
+		return
+	}
+
+	var skipCertVerify bool
+	if skipCertVerify, err = pConf.FieldBool(soFieldSkipCertVerify); err != nil {
+		return
+	}
+
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	tr.TLSClientConfig.InsecureSkipVerify = skipCertVerify
+	o.client = http.Client{Transport: tr}
+
+	return
+}
+
+//------------------------------------------------------------------------------
+
+func (o *output) Connect(_ context.Context) error { return nil }
+
+func (o *output) WriteBatch(ctx context.Context, b service.MessageBatch) (err error) {
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Set("Authorization", "Splunk "+o.token)
+
+	var payload bytes.Buffer
+	var payloadWriter io.Writer = &payload
+	var gzipFlusher func() error
+	if o.useGzipCompression {
+		header.Set("Content-Encoding", "gzip")
+		gzipper := gzip.NewWriter(&payload)
+		payloadWriter = gzipper
+		gzipFlusher = gzipper.Close
+	}
+	encoder := json.NewEncoder(payloadWriter)
+
+	for _, msg := range b {
+		data, err := msg.AsStructuredMut()
+		if err != nil {
+			rawData, err := msg.AsBytes()
+			if err != nil {
+				return fmt.Errorf("failed to get message bytes: %s", err)
+			}
+			data = map[string]any{"event": string(rawData)}
+		}
+
+		var dataObj map[string]any
+		var ok bool
+		if dataObj, ok = data.(map[string]any); !ok {
+			dataObj = map[string]any{"event": data}
+		} else if _, ok := dataObj["event"]; !ok {
+			dataObj = map[string]any{"event": data}
+		}
+
+		if o.eventHost != "" {
+			dataObj["host"] = o.eventHost
+		}
+		if o.eventSource != "" {
+			dataObj["source"] = o.eventSource
+		}
+		if o.eventSourceType != "" {
+			dataObj["sourcetype"] = o.eventSourceType
+		}
+		if o.eventIndex != "" {
+			dataObj["index"] = o.eventIndex
+		}
+
+		err = encoder.Encode(dataObj)
+		if err != nil {
+			return fmt.Errorf("failed to marshal message to json: %s", err)
+		}
+	}
+
+	if o.useGzipCompression {
+		if err := gzipFlusher(); err != nil {
+			return fmt.Errorf("failed to compress messages: %s", err)
+		}
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, o.url, &payload)
+	if err != nil {
+		return fmt.Errorf("failed to construct http request: %s", err)
+	}
+	req.Header = header
+	req.ContentLength = int64(payload.Len())
+
+	resp, err := o.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to execute http request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	// TODOOOOOOOOOOOOOOOOOOOOOOOOO ERROR HANDLING
```

Comment:
As above, I think more granular error handling is great but this is fine for now.